plugin.go

```diff
@@ -398,37 +412,63 @@ func (p *Plugin) Exec() error {
-		NetworkConfiguration: p.setupServiceNetworkConfiguration(),
-	}
+	taskDefinition := *(resp.TaskDefinition.TaskDefinitionArn)
 
-	if p.DesiredCount >= 0 {
-		sparams.DesiredCount = aws.Int64(p.DesiredCount)
-	}
 
+/// New section
 
-/// New section - run task
-    //fmt.Println("Task params:")
     taskParams := &ecs.RunTaskInput{
         Cluster:                    aws.String(p.Cluster),
-        Count:                      aws.Int64(1), // TODO: specify
-        //EnableExecuteCommand:     true, //?
+        Count:                      aws.Int64(p.DesiredCount),
         Group:                      aws.String(p.Family),
         LaunchType:                 aws.String(p.Compatibilities),
         NetworkConfiguration:       p.setupServiceNetworkConfiguration(),
-        //Overrides:                TODO
-        //PlacementConstraints:     TODO
+        //Overrides:                TODO, or not - why override container if everything defined in this run (to consider)
         //PlacementStrategy:        TODO
-        PlatformVersion:            aws.String("LATEST"), // TODO: specify
-        PropagateTags:              aws.String("TASK_DEFINITION"), // TODO: specify // values: TASK_DEFINITION or not specified
         //Tags:                     TODO
-        TaskDefinition:             aws.String(val), // TODO: give an option to not create a new task definition on each run, use existing one instead
+        TaskDefinition:             aws.String(taskDefinition), // TODO: give an option to not create a new task definition on each run, use existing one instead (to consider)
+        EnableExecuteCommand:       aws.Bool(p.EnableExecuteCommand),
+    }
+
+    if (p.Compatibilities == "FARGATE") && (len(p.PlatformVersion) > 0) {
+        taskParams.PlatformVersion = &p.PlatformVersion
+    }
+    if (p.PropagateTags) {
+        taskParams.PropagateTags = aws.String("TASK_DEFINITION")
+    }
+
+    for _,capElem := range p.CapacityProviders {
+        cleanedCap := strings.Trim(capElem, " ")
+	    parts := strings.SplitN(cleanedCap, " ", 3)
+	    base, baseErr := strconv.ParseInt(parts[0], 10, 64)
+	    weight, weightErr := strconv.ParseInt(parts[1], 10, 64)
+	    if baseErr != nil {
+			baseWrapperErr := errors.New(BaseParseErr + baseErr.Error())
+			LogTime()
+			fmt.Println(baseWrapperErr.Error())
+			return baseWrapperErr
+		}
+	    if weightErr != nil {
+			weightWrappedErr := errors.New(WeightParseErr + weightErr.Error())
+			LogTime()
+			fmt.Println(weightWrappedErr.Error())
+			return weightWrappedErr
+		}
+	    cap := &ecs.CapacityProviderStrategyItem {
+	        Base:               &base,
+	        Weight:             &weight,
+	        CapacityProvider:   &parts[2],
+	    }
+	    taskParams.CapacityProviderStrategy = append(taskParams.CapacityProviderStrategy, cap)
     }
-    //fmt.Println(taskParams)
+
+    // Standalone task's placementConstraints seems to be redundant if we already use them in task definition... yes/no?
```

Comment:
Remove it if not in use?

Reply:
I'd leave this comment to keep it in back of the head. And we'll see whether it's needed when we go live with this plugin.